generate_readme.py

```diff
@@ -0,0 +1,117 @@
+import json
+import os
+import warnings
+from typing import Union
+
+
+def get_info(filename: str):
+
+    with open(filename) as f:
+        data = json.load(f)
+
+    if not isinstance(data["topology"], list):
+        raise RuntimeError(
+            f"The topology is expected to be a list of edges (List[List[int, int]] or List[List[str, str]]), but received a {type(data['topology'])}."
+        )
+
+    info = {key: data[key] for key in ("nqubits", "qubits", "topology")}
+    one_q_native_gates = list(
+        next(iter(data["native_gates"]["single_qubit"].values())).keys()
+    )
+    two_q_native_gates = list(
+        next(iter(data["native_gates"]["two_qubit"].values())).keys()
+    )
+    info.update(
+        {
+            "native_gates": {
+                "single_qubit": one_q_native_gates,
+                "two_qubit": two_q_native_gates,
+            }
+        }
+    )
+    return info
+
+
+def create_mermaid_graph(
+    nodes: Union[list[str], list[int]], topology: Union[list[str], list[int]]
+):
+    node_to_index = dict(zip(nodes, range(len(nodes))))
+    if isinstance(nodes[0], int) or nodes[0].isdecimal():
+        edges = "\n    ".join(
+            [
+                f"{edge[0]}(({str(edge[0])})) <--> {edge[1]}(({str(edge[1])}));"
+                for edge in topology
+            ]
+        )
+    else:
+        edges = "\n    ".join(
+            [
+                f"{node_to_index[edge[0]]}(({edge[0]})) <--> {node_to_index[edge[1]]}(({edge[1]}));"
+                for edge in topology
+            ]
+        )
+    markdown_str = f"""
+```mermaid
+---
+config:
+layout: elk
+---
+graph TD;
+    {edges}
+```
+"""
+    return markdown_str
+
+
+def create_readme(filename: str):
+    info = get_info(filename)
+    mermaid_graph = create_mermaid_graph(info["qubits"], info["topology"])
+    qubits = (
+        ", ".join([str(q) for q in info["qubits"]])
+        if isinstance(info["qubits"][0], int) or info["qubits"][0].isdecimal()
+        else ", ".join([f"{q} ({i})" for i, q in enumerate(info["qubits"])])
+    )
+    readme_str = f"""
+## Native Gates
+**Single Qubit**: {", ".join(info["native_gates"]["single_qubit"])}
+
+**Two Qubit**: {", ".join(info["native_gates"]["two_qubit"])}
+
+## Topology
+**Number of qubits**: {info["nqubits"]}
+
+**Qubits**: {qubits}
+{mermaid_graph}
+"""
+    return readme_str
+
+
+if __name__ == "__main__":
+
+    platforms = [
+        directory.name
+        for directory in os.scandir(".")
+        if directory.is_dir()
+        and not (directory.name.startswith("_") or directory.name.startswith("."))
```

Comment:
```suggestion
        for directory in Path().glob("[!._]*/")
```
importing `Path` from `pathlib` on top.

Reply:
Uhm...

```py
[ins] In [2]: list(Path().glob("[!._]*/"))
Out[2]:
[PosixPath('qw11qD'),
 PosixPath('qw5q_platinum'),
 PosixPath('qw11q'),
 PosixPath('iqm5q')]
```

Reply:
Are you sure you didn't forget the final `/`?

Reply:
Also tested on macOS now:
```py
[ins] In [2]: list(Path().glob("[!._]*/"))
Out[2]:
[PosixPath('tests'),
 PosixPath('examples'),
 PosixPath('doc'),
 PosixPath('htmlcov'),
 PosixPath('src')]
                                                                                                                                                                                                                
[ins] In [3]: list(Path().glob("[!._]*"))
Out[3]:
[PosixPath('CODE_OF_CONDUCT.md'),
 PosixPath('flake.lock'),
 PosixPath('selfhosted'),
 PosixPath('LICENSE'),
 PosixPath('pyproject.toml'),
 PosixPath('tests'),
 PosixPath('tags'),
 PosixPath('README.md'),
 PosixPath('flake.nix'),
 PosixPath('examples'),
 PosixPath('doc'),
 PosixPath('coverage.xml'),
 PosixPath('poetry.lock'),
 PosixPath('htmlcov'),
 PosixPath('src')]
```

Which Python version are you using?

Reply:
3.10.15

Reply:
Found https://docs.python.org/3/library/pathlib.html#pathlib-pattern-language

> Changed in version 3.11: Globbing with a pattern that ends with a pathname components separator ([sep](https://docs.python.org/3/library/os.html#os.sep) or [altsep](https://docs.python.org/3/library/os.html#os.altsep)) returns only directories.

Reply:
The patch could be reverting to the previous behavior, for compatibility:
```py
        for directory in Path().glob("[!._]*/") if directory.is_dir()
````